docs/api_docs.rst

```diff
@@ -0,0 +1,316 @@
+    from xaitk_saliency import GenerateClassifierConfidenceSaliency
+    from xaitk_saliency.utils.masking import occlude_image_batch
+    from MyIntegration import classify_images  # type: ignore
+
+
+    # Pretend we have implementations of the standard interfaces.
+    class PerturbImageImplementation (PerturbImage):
+      ...
+
+
+    class GenerateClassifierConfidenceSaliencyImplementation (GenerateClassifierConfidenceSaliency):
+      ...
+
+
+    # Initializing an implementation of perturbation-based algorithms
+    perturb_image = PerturbImageImplementation()
+
+    # Initializing an implementation of classifier-based saliency generator
+    classifier_saliency = GenerateClassifierConfidenceSaliencyImplementation()
+
+    ...
+
+    # Loading reference image from file
+    ref_image = np.asarray(PIL.Image.open("some/test/image.png"))
+
+    # Generate perturbed images and perturbation masks on
+    # reference image on which saliency needs to be computed
+    mask_array = perturb_image(ref_image)
+    perturbed_images = occlude_image_batch(ref_image, mask_array)
+
+    # Compute class confidence predictions for reference and perturbed images.
+    # We assume for this example that this black-box image classification function
+    # returns a matrix of class label confidences with different class labels
+    # corresponding to different columns of the output matrix, whose shape will be
+    # [nInputs x nClasses].
+    ref_class_confs = classify_images([ref_image])[0]
+    perturbed_class_confs = classify_images(perturbed_images)
+
+    # We will also show the example case where we do not want to pass along all
+    # class confidences for saliency map generation, but only a select few.
+    # Maybe this would be defined by some interface or configuration.
+    pertinent_class_indices = [1, 4, 10]
+    ref_class_confs2 = ref_class_confs[pertinent_class_indices]
+    perturbed_class_confs2 = perturbed_class_confs[..., pertinent_class_indices]
+
+    # Computing the final classifier-based saliency map using
+    # classifier confidence on the original feature vector of an reference image
+    # along with the classifier confidence on all descriptors computed on the
+    # perturbed versions of the reference image and masks used to perturb the reference
+    # image
+    classifier_saliency_map = classifier_saliency(
+      ref_class_confs2,  # shape: [len(pertinent_class_indices)]
+      perturbed_class_confs2,  # shape: [len(perturbed_images), len(pertinent_class_indices)]
+      mask_array  # shape: [len(perturbed_images), ref_image.height, ref_image.width]
+    )
+    # There should be an equal number of saliency maps output as the number of
+    # distinct class confidences input:
+    assert len(classifier_saliency_map) == len(pertinent_class_indices)
+    # The shape of the output heatmap should be congruent to the shape of input
+    # perturbation masks.
+    assert classifier_saliency_map[0].shape == mask_array[0].shape
```

Comment:
Given D-RISE has now been merged, did we want to create an example for the GenerateDetectorProposalSaliency interface? Obviously, this could be done in another PR as well since the code example for this doesn't yet exist in the Google Doc.

Reply:
I would call this a non-trivial addition so I would call it for another PR. It may be the case that, now that we're in the actual repository, to reference a notebook or something that is testable via CI. Otherwise our "code" documentation can become stale.